TEST_APPS/device/nfcapp/nfcprocessCtrl.cpp

```diff
@@ -0,0 +1,195 @@
+     *
+     * @return NFC_OK in case of success or a meaningful error code in case of
+     * failure.
+     */
+    nfc_err_t NFCProcessController::start_discovery() {
+    	cmd_printf("start_discovery()\r\n");
+
+    	nfc_err_t err = _nfc_controller.start_discovery();
+
+        return(err);
+    }
+
+    /**
+     * Stop discovery.
+     *
+     * @return NFC_OK in case of success or a meaningful error code in case of
+     * failure.
+     */
+    nfc_err_t NFCProcessController::stop_discovery() {
+        cmd_printf("stop_discovery()\r\n");
+        return _nfc_controller.cancel_discovery();
+    }
+
+    nfc_rf_protocols_bitmask_t NFCProcessController::get_rf_protocols()
+    {
+        cmd_printf("get_supported_rf_protocols()\r\n");
+        _nfc_controller.set_delegate(this);
+        return _nfc_controller.get_supported_rf_protocols();
+    }
+
+
+    nfc_err_t NFCProcessController::set_rf_protocols(nfc_rf_protocols_bitmask_t protocols)
+    {
+        cmd_printf("configure_rf_protocols()\r\n");
+
+        return _nfc_controller.configure_rf_protocols(protocols);
+    }
+
+    /* ------------------------------------------------------------------------
+     * Implementation of NFCRemoteInitiator::Delegate
+     */
+    void NFCProcessController::on_connected() {
+        cmd_printf("on_connected()\r\n");
+    }
+
+    void NFCProcessController::on_disconnected() {
+        cmd_printf("on_disconnected()\r\n");
+
+        // reset the state of the remote initiator
+        _nfc_remote_initiator->set_delegate(NULL);
+        _nfc_remote_initiator.reset();
+
+        // restart peer discovery
+        _nfc_controller.start_discovery();
+    }
+
+    void NFCProcessController::parse_ndef_message(const Span<const uint8_t> &buffer) {
+        size_t len = buffer.size();
+        // copy remotely written message into our dummy buffer
+        if (len <= sizeof(_ndef_write_buffer)) {
+            cmd_printf("Store remote ndef message of size %d\r\n", len);
+            memcpy(_ndef_write_buffer, buffer.data(), len);
+            _ndef_write_buffer_used = len;
+        } else {
+            cmd_printf("Remote ndef message of size %d too large!\r\n", len);
+        }
+    }
+
+    size_t NFCProcessController::build_ndef_message(const Span<uint8_t> &buffer) {
+        cmd_printf("Copying message %d bytes to query buffer\r\n", _ndef_write_buffer_used);
+        memcpy(buffer.data(), _ndef_write_buffer, _ndef_write_buffer_used);
+        for (size_t k=0 ; k<_ndef_write_buffer_used; k++ ) {
+        	cmd_printf("%02x ", buffer.data()[k] );
+        }
+        return _ndef_write_buffer_used;
+    }
+
+    const char *NFCProcessController::str_discovery_terminated_reason(nfc_discovery_terminated_reason_t reason)
+    {
+        static const char* reasons[4] = {"completed", "cancelled", "rf error", "unexpected!"};
+        switch (int(reason)) {
+        case 0 :
+        case 1 :
+        case 2 :
+            return (reasons[int(reason)]);
```

Comment:
you don't need to wrap the value in parens, you don't need to cast an enum into int